iroh-net/src/util.rs

```diff
@@ -29,3 +29,43 @@ impl<T> Drop for AbortingJoinHandle<T> {
         self.0.abort();
     }
 }
+
+impl<T> std::ops::Deref for AbortingJoinHandle<T> {
+    type Target = tokio::task::JoinHandle<T>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+/// Holds a handle to abort a task.
+///
+/// The task can be manually aborted, or aborted once self is dropped.
+/// See [`tokio::task::AbortHandle`].
+pub struct CancelOnDrop {
+    task_name: &'static str,
+    handle: tokio::task::AbortHandle,
+}
+
+impl CancelOnDrop {
+    pub fn new(task_name: &'static str, handle: tokio::task::AbortHandle) -> Self {
```

Comment:
ugh, TIL that `tokio::task::Builder` is unstable.  I swear I had used task names in the past...   Which is sad since tokio already has the concept of a task name, if only it was stable.  A `From` impl would have been nice, but I guess we can't without task names.